### PR TITLE
Close #20267: Filter out network errors from Nimbus

### LINKED ISSUE #20267: NimbusErrorException$RequestError: Network error: "fetch error: java.net.SocketTimeoutException"
https://sentry.prod.mozaws.net/operations/firefox-nightly/issues/11722256/?referrer=github_plugin

```
NimbusErrorException$RequestError: Error sending request: [no-sentry] Network error: Java error: "fetch error: java.net.SocketTimeoutException"
    at org.mozilla.experiments.nimbus.internal.NimbusError.intoException(nimbus.kt:14)
    at org.mozilla.experiments.nimbus.internal.NimbusClient.fetchExperiments(nimbus.kt:13)
    at org.mozilla.experiments.nimbus.Nimbus$fetchExperimentsOnThisThread$1.invoke(Nimbus.kt:1)
    at org.mozilla.experiments.nimbus.Nimbus.withCatchAll(Nimbus.kt:1)
    at org.mozilla.experiments.nimbus.Nimbus.fetchExperimentsOnThisThread$nimbus_release(Nimbus.kt:1)
...
(6 additional frame(s) were not displayed)

[INFO] Error sending request: [no-sentry] Network error: Java error: "fetch error: java.net.SocketTimeoutException"
```

### PR DESCRIPTION
**Pull request by jonalmeida**
We filter out the network request failures from Nimbus since there isn't anything we can do about it.

Filed an issue to upstream this change as well.

cc: @jhugman 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt [hunk 1]
```diff
@@ -96,3 +100,17 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
         NimbusDisabled()
     }
 }
+
+/**
+ * Classifies which errors we should forward to our crash reporter or not. We want to filter out the
+ * non-reportable ones if we know there is no reasonable action that we can perform.
+ *
+ * This fix should be upstreamed as part of: https://github.com/mozilla/application-services/issues/4333
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt:108]: Do we need a follow up issue to remove this comment when we do upstream this fix?

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt (hunk 1 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt:108]: Let's wait to see what the outcome of the discussion is in that ticket. There could be alternative solutions to this problem.

**Comment by jonalmeida** [comment]
Christian made a good point that the `isSentryEnabled` check is redundant, so we can remove that and simplify the logic a bit more.

**Review by csadilek — APPROVED** [review]
👍

**Comment by jonalmeida** [comment]
Failure from https://github.com/mozilla-mobile/fenix/issues/20320. 🥲

**Pull request merged**